docs/source/blogs/tech_blog/blog12_Inference_Time_Compute_Implementation_in_TensorRT-LLM.md

```diff
@@ -0,0 +1,425 @@
+
+
+### Core Features
+The core features including:
+
+
+Decouple inference-time compute method and execution backend. Provides `Controller` concept for users to define the method, `Worker` concept to develop execution backend and `ScaffoldingLlm` to provide API for users to integrate `Controller` and `Worker` and run the request. 
+
+
+Make the inference-time compute method modular and reusable. An inference time compute method can be
+composed of multiple modules. In scaffolding, `Controller` can be constructed by a series of `Sub-Controllers`, then users can flexibly assemble and replace the `Sub-Controllers`.
+
+
+Provides sufficient concurrency to achieve good performance while ease of use. Concurrency is the key for performance. `Scaffolding` provides three levels of concurrency. The first level is that the different requests to a `ScaffoldingLlm` instance can be concurrent. The second level is that the multiple `Sub-Controllers` can be concurrent.The third level is that the multiply Tasks which yielded from `Controller` can be concurrent.
+
+
+### Architecture
+`Scaffolding` consists of three core components. Let's first briefly introduce these components. The `Worker` class is the backend that execute a single task, such as sending an inference request to an LLM inference framework or service, or completing a call to an external tool. The `Controller` class focuses on defining the workflow of a inference-time compute method. The `ScaffoldingLlm` class is responsible for integrating the two and completing the entire task.
+
+
+This is the call sequence diagram of `Scaffolding`:
+<div align="center">
+    <img src="https://github.com/NVIDIA/TensorRT-LLM/raw/main/docs/source/blogs/media/tech_blog12_scaffolding_sequence.png" alt="Scaffolding Sequence" width="900px">
+</div>
+<p align="center"><sub><em>Figure 1. Scaffolding Sequence</em></sub></p>
+
+Here we can focus on two points. First, `ScaffoldingLlm` provides users with the interface. Second, the `Controller` does not directly call the Worker.
+
+
+Next, we will introduce the code of the core components.
+
+
+#### Worker
+```python
+class Worker(ABC):
+
+    async def run_task(self, task: Task) -> TaskStatus:
+        worker_cls = type(self)
+        if type(task) not in worker_cls.task_handlers:
+            return TaskStatus.WORKER_NOT_SUPPORTED
+        return await worker_cls.task_handlers[type(task)](self, task)
+
+    task_handlers = {}
+```
+`Worker`'s core interface is `run_task()`. This interface takes in a `Task`, then completes this `Task` and writes the result into the corresponding field. It should be noted that `run_task()` is an asynchronous function and it can be concurrently and asynchronously called with python asyncio.
```

Comment:
suggestion: "The core interface of `Worker` is `run_task()`, which accepts a `Task`, executes it, and writes the result to the appropriate field."